assets/app/scripts/directives/logViewer.js

```diff
@@ -269,6 +269,11 @@ angular.module('openshiftConsole')
                 streamer = null;
                 $scope.$evalAsync(function() {
                   $scope.autoScrollActive = false;
+                  // if no logs, they have already been archived.
+                  if(lastLineNumber === 0) {
+                    $scope.state = 'empty';
+                    $scope.emptyStateMessage = 'The logs are no longer available or could not be loaded.';
```

Comment:
Updated so the `.onError` message takes priority & `.onClose` will not set the `$scope.emptyStateMessage` if it has already been set (since error is more severe).  

Cancelled Build now shows this:

![build_cancel_sw_400_error](https://cloud.githubusercontent.com/assets/280512/13154245/1e096754-d646-11e5-8fac-1b800bc64fa2.png)